---
title: 'PRC2 and WT1 Mutations in TARGET AML'
author: "Jenny Smith"
date: "April 6, 2018"
output: html_document
---



```{r setup}
library(knitr)
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),tidy=TRUE, fig.align='center', fig.width = 10, fig.height = 10)
knitr::opts_knit$set(root.dir = '/fh/fast/meshinchi_s/workingDir/TARGET/AML_TARGET/RNA/mRNAseq/analysis/2018.04.09_PRC2_Expn_WT1Mutants/')
```


```{r message = FALSE, warning=FALSE}
library(stringr)
library(magrittr)
library(ggplot2)
library(dplyr)
library(tibble)
library(tidyr)
getwd()
```


#Read in the Clinical Data

```{r}
CDE.0531 <- read.csv("~/reference_mapping-files/TARGET_AML_current_asof_june30_2016_UPDATED_CLEAN_4.5.18.csv", 
                     stringsAsFactors = FALSE)

head(CDE.0531[,1:5])
```

```{r}
table(CDE.0531$WT1.mutation)
```


```{r}
CDE.1031 <- read.csv("~/reference_mapping-files/TARGET_AML_1031_CDE_cleaned_4.10.2018.csv", 
                     stringsAsFactors = FALSE)

head(CDE.1031[,1:5])
```


#Read in the expression data

```{r}
TPM.list <- readRDS("~/RNA_seq_Analysis/0000.00.03_Expression_Matrices/TARGET_AML_0531_1031_dupGenesRemoved_TPM_List.rds")

```


#Define Functions 

```{r}
GetGeneAlias <- function(genes,expnMat){
  #genes is a character vector of any length with the genes of interest to find their aliases. 
  #ExpnMat are the rownames of the expression matrix as genes and patients as columns
  
  library(magrittr)
  library()
  
  g.alias <- read.csv("~/RNA_seq_Analysis/0000.00.02_Reference_GeneInfo/Homo_sapiens.gene_info.alias.csv",
                    stringsAsFactors = FALSE)
  
  regex <- paste0("\\b",genes,"\\b") #\b in regex is for beginning or END of line or the beginning/end of a column/space
  
  aliases <- lapply(regex, function(x) grep( x, g.alias$alias, value=TRUE, ignore.case = TRUE)) %>%
    set_names(genes)
  
  indices <- lapply(names(aliases), 
                  function(x) which(rownames(expnMat) %in% unlist(strsplit(aliases[[x]], " ")))) %>%
    set_names(names(aliases))
  
  df <- do.call(rbind, lapply(indices, function(x) rownames_to_column(expnMat, "Gene")[x,]))
  
  res <- list("aliases"=aliases,"subset"=df)
  return(res)
}
```

```{r}
pheno_bars <- function(CDE,IDCol,cols){
  #CDE is the clinical data frame with patietns as rows. 
  #IDcol is the name of the column with patient USIs or COG#s
  #cols are the colnames to be combined. 
  
  replace_yes <- function(col,name){
    name <-gsub(".RNAseq|.positive.", "", name)
    col <- ifelse(col == "Yes", name, col)
    return(col)
  }
  
  colorbar.df <- CDE %>%
    select(IDCol,cols)
  
  if(length(cols) > 1){
    colorbar.df <- bind_cols(colorbar.df, mapply(replace_yes, CDE[,cols], cols, SIMPLIFY = FALSE))
  }else{
    new <- data.frame(replace_yes(CDE[,cols],cols)) %>% set_colnames(cols)
    colorbar.df <- bind_cols(colorbar.df, new) #dplyr bind_cols throws error Error in cbind_all(x) : Argument 2 must have names??
  }
  
  
  p <- NULL
  for (col in cols){p <- paste(p,colorbar.df[,paste0(col,1)], sep="_")}
  
  colorbar <- p %>%
    gsub("No|Unknown|_", "", .) %>%
    gsub("^$", "OtherAML",.) %>%
    set_names(CDE[,IDCol])
  
  return(colorbar)
  
}
```

#Description of PRC2 Genes 


Human PRC2 consists of four core subunits (Enhancer of zeste homolog 2 [EZH2], embryonic ectoderm development [EED], suppressor of zeste 12 [SUZ12], and retinoblastoma suppressor associated protein 46/48 [RbAp46/48]) and several auxiliary subunits including AE (adipocyte enhancer)-binding protein 2 [AEBP2], Jumonji/AT-rich interactive domain 2 [JARID2] and polycomb-like [PCL] proteins 1, 2, and 3]1. EZH2 is the enzymatic subunit of PRC2. The 751-amino-acid EZH2 protein harbors six domains: a WD-40 binding domain (WDB), domains I–II, two SWI3-ADA2-N-CoR-TFIIIB [SANT] domains, a cysteine-rich CXC domain and the evolutionarily conserved carboxy-terminal Su(var) 3–9-enhancer of zeste-trithorax (SET) methyltransferase domain. However, EZH2 lacks histone methyltransferase (HMTase) activity on its own and requires at least two other PRC2 core subunits, EED and SUZ12, for catalytic activity1,13,14.

doi:10.1038/aps.2017.7


Mutant WT1 is associated with DNA hypermethylation of PRC2 targets in AML and responds to EZH2 inhibition
Subarna Sinha, Daniel Thomas, Linda Yu, Andrew J. Gentles, Namyoung Jung, M. Ryan Corces-Zimmerman, Steven M. Chan, Andreas Reinisch, Andrew P. Feinberg, David L. Dill and Ravindra Majeti
Blood 2015 125:316-326; doi: https://doi.org/10.1182/blood-2014-03-566018


#Examine the Boxplots for each PRC2 Genes in 0531 

```{r}
PRC2.Genes <- c("EZH2", "EED", "SUZ12", "RbAp46", "RbAp48", "AEBP2", "JARID2", "PCL1", "PCL2")
```

```{r}
PRC2.Expn <- GetGeneAlias(PRC2.Genes, TPM.list$LD.0531)

PRC2.Expn$subset$Gene
```

```{r}
#same alias PCL1 but not related to PRC2 complex
PRC2.Expn$subset <- subset(PRC2.Expn$subset, Gene != "PCYOX1" & Gene != "PHF2") 

head(PRC2.Expn$subset[,1:5])
```


```{r}
ExpnByGroup <- PRC2.Expn$subset %>% 
  gather(TARGET.USI.1, TPM,-Gene) %>%
  spread(Gene,TPM) %>% 
  mutate(Group=ifelse(grepl("^BM", TARGET.USI.1), "NBM", "AML")) %>% 
  left_join(., select(CDE.0531,TARGET.USI.1, Age.at.Diagnosis.in.Days, WT1.mutation,FLT3.ITD.positive.), by="TARGET.USI.1") %>%
  select(TARGET.USI.1,Group, Age.at.Diagnosis.in.Days, WT1.mutation,FLT3.ITD.positive., everything()) %>%
  mutate_at(vars(3:5), funs(ifelse(grepl("^BM", TARGET.USI.1), "NBM", .))) %>%
  # mutate_at(vars(3:5), funs(ifelse(is.na(.), "Unknown", .)))
  filter(! is.na(WT1.mutation))
  

dim(ExpnByGroup)
# write.csv(ExpnByGroup, "TARGET_AML_0531_PRC2_Complex_Gene_Expression_Matrix_TPM.csv", row.names = FALSE)
```

```{r}
WT1.FLT3.ITD <- pheno_bars(CDE.0531, IDCol = "TARGET.USI.1", 
                           cols=c("WT1.mutation", "FLT3.ITD.positive.")) %>%
  as.data.frame() %>%
  rownames_to_column("TARGET.USI.1") %>%
  select(TARGET.USI.1,WT1.FLT3.ITD=".") %>%
  mutate(WT1.FLT3.ITD=gsub("WT1.mutationFLT3.ITD", "WT1.FLT3.ITD", WT1.FLT3.ITD)) %>%
  mutate(WT1.FLT3.ITD=gsub("WT1.mutation", "WT1.Alone", WT1.FLT3.ITD))

table(WT1.FLT3.ITD$WT1.FLT3.ITD)
```


```{r}
sum(subset(WT1.FLT3.ITD, WT1.FLT3.ITD == "WT1.Alone")$TARGET.USI.1 %in% colnames(PRC2.Expn$subset))
sum(subset(WT1.FLT3.ITD, WT1.FLT3.ITD == "WT1.FLT3.ITD")$TARGET.USI.1 %in% colnames(PRC2.Expn$subset))
sum(subset(CDE.0531,WT1.mutation == "Yes")$TARGET.USI.1  %in% colnames(PRC2.Expn$subset))
sum(subset(CDE.0531,WT1.mutation == "No")$TARGET.USI.1  %in% colnames(PRC2.Expn$subset))
```


```{r}
melted.expn <- PRC2.Expn$subset %>% 
  gather(TARGET.USI.1,TPM, -Gene) %>%
  mutate(Log2.TPM=log2(TPM+1)) %>%
  left_join(.,select(CDE.0531, TARGET.USI.1, WT1.mutation,FLT3.ITD.positive.), by=c("TARGET.USI.1")) %>%
  mutate(WT1.mutation=ifelse(grepl("^BM", TARGET.USI.1), "NBM", WT1.mutation)) %>%
  mutate(WT1.mutation=ifelse(is.na(WT1.mutation), "Unknown", WT1.mutation)) %>%
  filter(WT1.mutation != "Unknown") %>%
  left_join(., WT1.FLT3.ITD, by="TARGET.USI.1") %>%
  mutate(WT1.FLT3.ITD=ifelse(grepl("^BM", TARGET.USI.1), "NBM", WT1.FLT3.ITD))

head(melted.expn)
```


```{r}
# tiff("TARGET_AML_0531_PRC2_Gene_Expression_in_WT1_vs_OtherAMLs_boxplots.tiff", height = 10, width = 10, units="in", res=600)
ggplot(melted.expn, aes(x=WT1.mutation, y=Log2.TPM, fill=WT1.mutation)) + 
  geom_boxplot() + 
  facet_wrap(~Gene,ncol = 2, nrow = 5, scales = "free") + 
  theme_bw() + 
  scale_fill_brewer(palette = "Dark2")
# dev.off()
```


```{r}
melted.expn %>% 
  group_by(WT1.mutation,Gene) %>%
  summarise_at(vars("Log2.TPM"), funs(mean,median,max,min)) %>% 
  arrange(Gene)
```

```{r}
ggplot(melted.expn, aes(Log2.TPM, fill=WT1.mutation)) + 
  geom_density(alpha=0.5) +
  # geom_histogram(alpha=0.4, position ="identity", binwidth = 0.25) + 
  facet_wrap(~Gene,ncol = 2, nrow = 5, scales = "free") + 
  theme_bw() + 
  scale_fill_brewer(palette = "Dark2")
```


#Investigate Expression Differences within WT1 

```{r}
WT1.Only <- subset(melted.expn,WT1.mutation=="Yes")
```

```{r}
# tiff("TARGET_AML_0531_PRC2_Gene_Expression_in_WT1.FLT3.ITD_vs_WT1Alone_boxplots.tiff", height = 10, width = 10, units="in", res=600)
ggplot(WT1.Only, aes(x=WT1.FLT3.ITD, y=Log2.TPM, fill=WT1.FLT3.ITD)) + 
  geom_boxplot() + 
  facet_wrap(~Gene,ncol = 2, nrow = 5, scales = "free") + 
  theme_bw() + 
  scale_fill_brewer(palette="Set1")
# dev.off()
```

```{r}
Test.Sig <- WT1.Only %>%
  group_by(Gene) %>%
  spread(WT1.FLT3.ITD, Log2.TPM) %>%
  do(Wilcox=wilcox.test(.$WT1.Alone,.$WT1.FLT3.ITD,alternative = "two.sided"))

names(Test.Sig$Wilcox) <- Test.Sig$Gene

Test.Sig$Wilcox$EZH2
Test.Sig$Wilcox$RBBP4
Test.Sig$Wilcox$RBBP7
```


```{r}
x <- subset(WT1.Only, WT1.FLT3.ITD == "WT1.Alone" & Gene == "EZH2")$Log2.TPM
y <- subset(WT1.Only, WT1.FLT3.ITD == "WT1.FLT3.ITD" & Gene == "EZH2")$Log2.TPM

wilcox.test(x,y, alternative = "two.sided")
```



#Look at PCR2 gene expression in WT1 in 1031 

```{r}
PRC2.Expn.1031 <- GetGeneAlias(PRC2.Genes, TPM.list$HD.1031)
```

```{r}
PRC2.Expn.1031$subset <- PRC2.Expn.1031$subset %>%
  filter(! grepl("PCYOX1|PHF2", Gene))

head(PRC2.Expn.1031$subset[,1:5])
```


```{r}
ExpnByGroup.1031 <- PRC2.Expn.1031$subset %>% 
  gather(USI,TPM,-Gene) %>% 
  spread(Gene,TPM) %>%
  mutate(Group=ifelse(grepl("^BM|^RO", USI), "NBM", "AML")) %>% 
  left_join(., select(CDE.1031, USI, Age..Days.,WT1.mutation, FLT3.ITD), by="USI") %>% 
  select(USI,Group,Age..Days., WT1.mutation, FLT3.ITD, everything()) %>%
  mutate_at(vars(3:5), funs(ifelse(Group == "NBM", "NBM", .))) %>%
  filter(!is.na(Age..Days.))
  

head(ExpnByGroup.1031)
# write.csv(ExpnByGroup.1031, "TARGET_AML_1031_PRC2_Complex_Gene_Expression_Matrix_TPM.csv", row.names = FALSE)
```

```{r}
table(ExpnByGroup.1031$WT1.mutation, useNA = "always")
```


```{r}
WT1.FLT3.ITD_1031 <- pheno_bars(subset(CDE.1031, !is.na(CDE.1031$USI)), IDCol = "USI", 
                           cols=c("WT1.mutation", "FLT3.ITD")) %>%
  as.data.frame() %>%
  rownames_to_column("USI") %>%
  select(USI,WT1.FLT3.ITD=".") %>%
  mutate(WT1.FLT3.ITD=gsub("WT1.mutationFLT3.ITD|MaybeFLT3.ITD", "WT1.FLT3.ITD", WT1.FLT3.ITD)) %>%
  mutate(WT1.FLT3.ITD=gsub("Maybe", "WT1.Alone", WT1.FLT3.ITD)) %>%
  mutate(WT1.FLT3.ITD=gsub("WT1.mutation", "WT1.Alone", WT1.FLT3.ITD))



table(WT1.FLT3.ITD_1031$WT1.FLT3.ITD)
# table(WT1.FLT3.ITD_1031)
```

```{r}
melted.expn.1031 <- PRC2.Expn.1031$subset %>% 
  gather("USI", "TPM", -Gene) %>% 
  mutate(Group=ifelse(grepl("^BM|^RO", USI), "NBM","AML")) %>%
  left_join(., select(CDE.1031, USI, WT1.mutation, FLT3.ITD), by="USI") %>%
  left_join(., WT1.FLT3.ITD_1031, by="USI") %>%
  mutate_at(vars(5:7), funs(ifelse(Group == "NBM" & is.na(.), "NBM", ifelse(Group == "AML"& is.na(.), "Unknown", .)))) %>%
  mutate(log2.TPM=log2(TPM+1)) %>%
  filter(WT1.mutation != "Unknown") %>%
  mutate_at(vars("WT1.mutation"), funs(gsub("Maybe", "Yes", .))) %>% #make the Maybes Yes for increased #'s. Proportions look correct
  mutate_at(vars("WT1.mutation"), funs(factor(.,levels=c("NBM", "No", "Yes"))))
  
  
head(melted.expn.1031)
```


```{r fig.height=10, fig.width=10}
# tiff("TARGET_AML_1031_PRC2_Gene_Expression_in_WT1_vs_OtherAMLs_boxplots.tiff", height = 10, width = 10, units="in", res=600)

ggplot(melted.expn.1031, aes(x=WT1.mutation, y=log2.TPM, fill=WT1.mutation)) +
  geom_boxplot() + 
  facet_wrap(~Gene, ncol = 2, nrow = 5, scales = "free") + 
  theme_bw() + 
  scale_fill_brewer(palette="Dark2")

# dev.off()

```


```{r}
WT1.alone <- subset(melted.expn.1031, WT1.mutation=="Yes")
```

```{r fig.height=10, fig.width=10}

# tiff("TARGET_AML_1031_PRC2_Gene_Expression_in_WT1Alone_vs_WT1.FLT3.ITD_boxplots.tiff", height = 10, width = 10, units="in", res=600)

ggplot(WT1.alone, aes(x=WT1.FLT3.ITD, y=log2.TPM, fill=WT1.FLT3.ITD)) + 
  geom_boxplot()  + 
  facet_wrap(~Gene, ncol=2, nrow=5, scales="free") +
  theme_bw() +
  scale_fill_brewer(palette="Set1")

# dev.off()

```




#Session Information 

```{r}
sessionInfo()
```





















